src/dos/dos_keyboard_layout.cpp

```diff
@@ -292,37 +297,36 @@ Bitu keyboard_layout::read_keyboard_file(const char* keyboard_file_name, Bit32s
 	auto tempfile = OpenDosboxFile(nbuf);
 	if (tempfile==NULL) {
 		// try keyboard layout libraries next
-		if ((start_pos=read_kcl_file("keyboard.sys",keyboard_file_name,true))) {
-			tempfile = OpenDosboxFile("keyboard.sys");
-		} else if ((start_pos=read_kcl_file("keybrd2.sys",keyboard_file_name,true))) {
-			tempfile = OpenDosboxFile("keybrd2.sys");
-		} else if ((start_pos=read_kcl_file("keybrd3.sys",keyboard_file_name,true))) {
-			tempfile = OpenDosboxFile("keybrd3.sys");
-		} else if ((start_pos=read_kcl_file("keyboard.sys",keyboard_file_name,false))) {
-			tempfile = OpenDosboxFile("keyboard.sys");
-		} else if ((start_pos=read_kcl_file("keybrd2.sys",keyboard_file_name,false))) {
-			tempfile = OpenDosboxFile("keybrd2.sys");
-		} else if ((start_pos=read_kcl_file("keybrd3.sys",keyboard_file_name,false))) {
-			tempfile = OpenDosboxFile("keybrd3.sys");
-		} else if ((start_pos=read_kcl_data(layout_keyboardsys,33196,keyboard_file_name,true))) {
-			assert(read_buf_size == 0);
-			for (Bitu ct=start_pos+2; ct<33196; ct++) read_buf[read_buf_size++]=layout_keyboardsys[ct];
-		} else if ((start_pos=read_kcl_data(layout_keybrd2sys,25431,keyboard_file_name,true))) {
-			assert(read_buf_size == 0);
-			for (Bitu ct=start_pos+2; ct<25431; ct++) read_buf[read_buf_size++]=layout_keybrd2sys[ct];
-		} else if ((start_pos=read_kcl_data(layout_keybrd3sys,27122,keyboard_file_name,true))) {
-			assert(read_buf_size == 0);
-			for (Bitu ct=start_pos+2; ct<27122; ct++) read_buf[read_buf_size++]=layout_keybrd3sys[ct];
-		} else if ((start_pos=read_kcl_data(layout_keyboardsys,33196,keyboard_file_name,false))) {
-			assert(read_buf_size == 0);
-			for (Bitu ct=start_pos+2; ct<33196; ct++) read_buf[read_buf_size++]=layout_keyboardsys[ct];
-		} else if ((start_pos=read_kcl_data(layout_keybrd2sys,25431,keyboard_file_name,false))) {
-			assert(read_buf_size == 0);
-			for (Bitu ct=start_pos+2; ct<25431; ct++) read_buf[read_buf_size++]=layout_keybrd2sys[ct];
-		} else if ((start_pos=read_kcl_data(layout_keybrd3sys,27122,keyboard_file_name,false))) {
+		auto try_file = [&](const char* file, bool first_id_only) {
+			if (!(start_pos=read_kcl_file(file,keyboard_file_name,first_id_only)))
+				return false;
+			tempfile = OpenDosboxFile(file);
+			return true;
+		};
+		auto try_builtin = [&](uint8_t i, bool first_id_only) {
+			if (!(start_pos=read_kcl_data(BLOB_KEYBOARD_SYS[i],keyboard_file_name,first_id_only)))
+				return false;
```

Comment:
:+1: , very nice collapsing of a lot of duplicating code!

Sooo much nicer that the old stuff it replaced.  Wow.